actgen/agents/action_dqn.py

```diff
@@ -5,37 +5,37 @@
 
 
 class ActionDQNAgent(DQNAgent):
-    def __init__(self, env, params):
-        super().__init__(env, params)
+    def __init__(self, observation_space, action_space, get_duplicate_actions, params):
+         super().__init__(observation_space, action_space, get_duplicate_actions, params)
 
     def _get_q_targets(self, batch):
         with torch.no_grad():
             # Compute Double-Q targets
-            next_states = torch.stack(batch.next_state, dim=0).float()
+            next_states = torch.stack(batch.next_state, dim=0).float().to(self.params['device'])
             aps = []
             for s in next_states:
                 q_values = self._get_q_values_for_state(s).squeeze().float()
                 ap = torch.argmax(q_values, dim=0)
                 aps.append(ap)
-            aps = torch.as_tensor(aps)
+            aps = torch.as_tensor(aps).to(self.params['device'])
```

Comment:
I'm surprised we need a `to(device)` here. Should this be a `stack` or `concat` instead of an `as_tensor`? Each `ap` should have the same device as `q_values`, which should inherit whatever device `s` and the model use.